Makefile.common

```diff
@@ -0,0 +1,3 @@
+# Points to CHARM_HOME
+CHARMC=/home/nk/src/charm/bin/charmc
+HYPERCOMM_HOME=/home/nk/src/hypercomm
```

Comment:
These are user-specific paths -- how can they be generalized?

Reply:
Wait this file should not exist in the PR. This file is generated from `Makefile.common.in` using cmake's `configure_file` facility. Let me correct it.

As for generalizing it: see https://github.com/jszaday/hypercomm/blob/cmake/CMakeLists.txt#L26. This takes the template file and generates user-specific paths.